src/libraries/System.Private.CoreLib/src/System/HashCode.cs

```diff
@@ -319,6 +320,38 @@ public void AddBytes(ReadOnlySpan<byte> value)
             ref byte pos = ref MemoryMarshal.GetReference(value);
             ref byte end = ref Unsafe.Add(ref pos, value.Length);
 
+            if ((nint)Unsafe.ByteOffset(ref pos, ref end) < sizeof(int) * 4)
```

Comment:
For .NET 7/C# 11, we can avoid the cast since `nint`/`IntPtr` are now true aliases of eachother.
```suggestion
            if (Unsafe.ByteOffset(ref pos, ref end) < (sizeof(int) * 4))
```